### ISSUE TITLE
Remember "fold" status for open files

### ISSUE THREAD

**Issue by ghost** [geany/geany#793]
Hello

For example, If I open a geany project with 2 documents.
- I click on "document » fold all"
- I click on "project » close"

Now if I reopen the same project, "fold" status are lost, I need to click again on "document » fold all".
Please, can you save fold status?

Thanks (Geany 1.26)

**Comment by elextr** [comment]
Saving fold status has been discussed before, but nobody came up with an acceptable solution to how to detect and handle the situation of the file having changed since the fold status was saved.  Concrete proposals are welcome.

**Comment by JanAxelsson** [comment]
I think the stored fold-state deserves being in standard Geany, and not relying on a plugin.

**From previous post, above :**
_"Saving fold status has been discussed before, but nobody came up with an acceptable solution to how to detect and handle the situation of the file having changed since the fold status was saved. Concrete proposals are welcome."_

**Proposal** :
I don't know if you find this solution feasible, but wouldn't it be possible to :
A) keep a memory of the fold state (in a file in the Geany folder?) together with 
B) a record of modified date and time of the file ? 

**How would you detect the situation of file changed since the fold status** ?
Compare stored fold-time with file modification-time
 
**How would you handle if an opened file was changed since last stored fold-state ?**  
I would find it acceptable resetting the fold-state -- it is still a great improvement from how it works now

**Comment by elextr** [comment]
@JanAxelsson at first glance that may work, but by "concrete" I also meant a pull request :grin:.

I would suggest saving it in the session, ie the users `geany.conf` or project file, whichever is in use.

**Comment by hyoretsu** [comment]
Any chance this gets implemented? Or is there a plugin that does this? A simple solution would be store folds and a cache of the file or something, a hash, if anything changed externally discard the folds.

**Comment by b4n** [comment]
I think there's a plugin that doors this, maybe it is numbered bookmarks or something

**Comment by eht16** [comment]
Yes, it's https://plugins.geany.org/geanynumberedbookmarks.html.
